src/QMCDrivers/VMC/VMCUpdateAll.cpp

```diff
@@ -35,139 +35,55 @@ VMCUpdateAll::~VMCUpdateAll()
 
 void VMCUpdateAll::advanceWalker(Walker_t& thisWalker, bool recompute)
 {
-    makeGaussRandomWithEngine(deltaR,RandomGen);
-    //if (!W.makeMove(thisWalker,deltaR, m_sqrttau))
-    if (!W.makeMove(thisWalker,deltaR,SqrtTauOverMass))
-    {
-      H.rejectedMove(W,thisWalker);
-      return;
-    }
-    //W.R = m_sqrttau*deltaR + thisWalker.R;
-    //W.update();
-    RealType logpsi(Psi.evaluateLog(W));
-   // app_log()<<logpsi<< std::endl;
-    RealType g= std::exp(2.0*(logpsi-thisWalker.Properties(LOGPSI)));
-    if (RandomGen() > g)
-    {
-      thisWalker.Age++;
-      ++nReject;
-      H.rejectedMove(W,thisWalker);
+  /* thisWalker.R will track the last accepted configuration
+   * W.R will track the proposed configuration 
+   *
+   * upon Call:
+   *   thisWalker.R,G,L,DistTables,SK,Properties must be consistent
```

Comment:
I think thisWalker doesn't carry DistanceTables and SK.

Reply:
@ye-luo that is a good point, I was confused. 

It seems that ParticleSet::loadWalker transfers R,G,L from thisWalker to W, but only update W.DistTables or SK for pbyp driver. I think this means the internal states of W become inconsistent after calling saveWalker in the wbyw driver. 

If there a reason why the `if (pbyp)` has to be in `ParticleSet::loadWalker`?

Reply:
This is necessary.
In pbyp moves, e-I distance table need to be up-to-date only when JeeI is used. SK need to be up-to-date if SK is needed at every pbyp move.
In allp moves, nothing is needed. Only update distance table and SK for proposed moves, if the last move is rejected, update everything with the old configuration before Hamiltonian and estimators.

At this line, please say `thisWalker.R,G,Properties must be consistent`